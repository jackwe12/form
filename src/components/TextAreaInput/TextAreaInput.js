import React from 'react';
import ValidationError from '../ValidationError/ValidationError';

function TextareaInput({ name, label, type, value, alert, setInputs }) {
    return (
        <div>
            <label htmlFor={name} className="uk-form-label uk-text-large">{label}</label>
            <textarea id={name} name={name} value={value} type={type} onChange={setInputs} className={"uk-textarea" + (alert ? ' uk-form-danger' : '')}></textarea>
            <ValidationError content={alert} />
        </div>
    )
}

export default TextareaInput;