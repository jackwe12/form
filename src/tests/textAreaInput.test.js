import TextAreaInput from '../components/TextAreaInput/TextAreaInput';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


const mockHandler = jest.fn();
const wrapper = shallow(
    <TextAreaInput 
        setInputs={mockHandler}
        {...{
        name: "description",
        label: "Description",
        alert: "",
        }
    }

/>)
const wrapper2 = shallow(
    <TextAreaInput 
        setInputs={mockHandler}
        {...{
        name: "description",
        label: "Description",
        alert: "error",
        }
    }

/>)

describe('Testing <TextAreaInput/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
        .create(      <TextAreaInput 
            setInputs={mockHandler}
            {...{
            name: "description",
            label: "Description",
            alert: "error",
            }
        }
    
    />).toJSON();
            expect(tree).toMatchSnapshot();
        })
    test('label tag renders correct text',()=>{
        expect(wrapper.find('label').text()).toEqual('Description');
    });

    test('textarea tag renders correct prop',()=>{
        expect(wrapper.find('textarea').prop('name')).toEqual('description');
        expect(wrapper.find('textarea').prop('id')).toEqual('description');
        expect(wrapper.find('textarea').prop('value')).toEqual(undefined);

    });
    test('should call mock changeHandler when textarea changes',()=>{
        const textarea = wrapper.find('textarea');
        textarea.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
    test('when alert is empty, no uk-form-danger class name',()=>{
        expect(wrapper.find('textarea').hasClass("uk-form-danger")).toBe(false);       
    });
    test('when alert exists, uk-form-danger class name exists',()=>{
        expect(wrapper2.find('textarea').hasClass("uk-form-danger")).toBe(true);       

    });
});
