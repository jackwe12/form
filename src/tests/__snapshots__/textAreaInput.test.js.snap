// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TextAreaInput/> snapshot testing 1`] = `
<div
  className="input-component"
>
  <label
    className="form-label"
  >
    Description
  </label>
  <textarea
    className="uk-textarea uk-form-danger"
    id="description"
    name="description"
    onChange={[MockFunction]}
  />
  <div
    className="uk-text-danger"
    style={
      Object {
        "height": "21px",
      }
    }
  >
    <img
      alt="exclamation-circle.png"
      className="exclamation-circle"
      onClick={[Function]}
      src="exclamation-circle.png"
    />
    
  </div>
</div>
`;
